website/events/api/v2/admin/views.py

```diff
@@ -0,0 +1,149 @@
+from events.api.v2.admin import filters
+from events.api.v2.admin.permissions import IsOrganiser
+from events.api.v2.admin.serializers.event import EventListSerializer, EventSerializer
+from events.api.v2.admin.serializers.event_registration import (
+    EventRegistrationSerializer,
+)
+from events.models import Event, EventRegistration
+from thaliawebsite.api.v2.admin.views import (
+    AdminListAPIView,
+    AdminRetrieveAPIView,
+    AdminCreateAPIView,
+    AdminUpdateAPIView,
+    AdminDestroyAPIView,
+)
+import events.api.v2.filters as normal_filters
+
+
+class EventAdminListCreateAPIView(AdminListAPIView, AdminCreateAPIView):
+    queryset = Event.objects.prefetch_related("organiser")
+    permission_classes = [IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["events:admin"]
+    filter_backends = [
+        framework_filters.OrderingFilter,
+        normal_filters.CategoryFilter,
+        normal_filters.OrganiserFilter,
+        normal_filters.EventDateFilter,
+        filters.PublishedFilter,
+    ]
+    ordering_fields = (
+        "start",
+        "end",
+        "published",
+        "registration_start",
+        "registration_end",
+    )
+
+    def get_serializer_class(self):
+        if self.request.method.lower() == "post":
+            return EventSerializer
+        return EventListSerializer
+
+
+class EventAdminDetailAPIView(
+    AdminRetrieveAPIView, AdminUpdateAPIView, AdminDestroyAPIView
+):
+    queryset = Event.objects.all()
+    serializer_class = EventSerializer
+    permission_classes = [IsOrganiser, IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["events:admin"]
+
+
+class EventRegistrationAdminListView(AdminListAPIView, AdminCreateAPIView):
+    """Returns a list of registrations."""
+
+    serializer_class = EventRegistrationSerializer
+    permission_classes = [IsOrganiser, IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["events:admin"]
+    filter_backends = (
+        framework_filters.OrderingFilter,
+        filters.EventRegistrationCancelledFilter,
+    )
+    ordering_fields = ("queue_position", "date", "date_cancelled")
+
+    def get_queryset(self):
+        event = get_object_or_404(Event, pk=self.kwargs.get("pk"))
+        if event:
+            return EventRegistration.objects.filter(event_id=event).prefetch_related(
+                "member", "member__profile"
+            )
+        return EventRegistration.objects.none()
+
+
+class EventRegistrationAdminDetailView(AdminRetrieveAPIView, AdminUpdateAPIView):
+    """Returns details of an event registration."""
+
+    serializer_class = EventRegistrationSerializer
+    queryset = EventRegistration.objects.all()
+    permission_classes = [IsOrganiser, IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["events:admin"]
+    event_lookup_field = "event_id"
+
+    def get_queryset(self):
+        return super().get_queryset().filter(event=self.kwargs["event_id"])
+
+
+class EventRegistrationAdminFieldsView(APIView):
```

Comment:
This should be an AdminAPIView I think?

also maybe the `get` `put` and `patch` methods should call their `super()` methods

Reply:
There are no super methods. But I had to include the permission mixin indeed.